docs/resources/stream_on_directory_table.md

```diff
@@ -72,6 +72,7 @@ resource "snowflake_stream_on_directory_table" "stream" {
 - `fully_qualified_name` (String) Fully qualified name of the resource. For more information, see [object name resolution](https://docs.snowflake.com/en/sql-reference/name-resolution).
 - `id` (String) The ID of this resource.
 - `show_output` (List of Object) Outputs the result of `SHOW STREAMS` for the given stream. (see [below for nested schema](#nestedatt--show_output))
+- `stale` (Boolean) Indicated if the stream is stale. When Terraform detects that the stream is stale, the stream is recreated with `CREATE OR REPLACE`.
```

Comment:
From the documentation page you sent me, it seems you can just call https://docs.snowflake.com/en/sql-reference/functions/system_stream_has_data to prevent the stream from becoming stale instead of recreating the whole stream, but maybe eventually we would have to re-create it anyway (?). Not sure, but we could check if that would work instead of re-creating the stream.

Reply:
It can be used to extend stale_after date on already non-stale streams. When you call this function on a stale stream, it doesn't make it non-stale.